aiida/orm/nodes/node.py

```diff
@@ -47,9 +49,16 @@ def __init__(self, fileobj, name):
     def _warn_if_not_entered(self, method):
         """Fire a warning if the object wrapper has not yet been entered."""
         if not self._was_entered:
-            msg = '`{}` used without context manager for {}. This will raise starting from `aiida-core==2.0.0`'.format(
-                method, self._name
-            )
+            msg = f'\n`Node.{method}` used outside context manager for {self._name}.\n' + \
+                  'Please use `with <node>.open(): ...` instead. ' + \
+                  'This will raise starting from `aiida-core==2.0.0`\n'
+            caller_frame = inspect.currentframe().f_back.f_back
```

Comment:
I wonder if there is a possibility that this code fails - maybe we can just wrap it in a try / except such that it fails gracefully?

Reply:
Done (for the case when `f_back` returns `None`